types/clevertap/index.d.ts

```diff
@@ -0,0 +1,168 @@
+
+interface UploadOptions {
+    debug?: number;
+    batchSize?: number;
+}
+
+interface Query {
+    event_name: string;
+    props?: Array<Record<string, any>>;
+    from: number;
+    to: number;
+}
+
+interface QueryOptions {
+    debug?: number;
+    batchSize?: number;
+}
+
+type ProfileIdentity = RequireAtLeastOne<{
+    email?: string;
+    identity?: string;
+    objectId?: string;
+}>;
+
+type ProfileOptions = {
+    debug?: number;
+} & ProfileIdentity;
+
+interface CustomControlGroup {
+    type: 'custom';
+    name: string;
+}
+
+interface CampaignControlGroup {
+    type: 'campaign';
+    percentage: number;
+}
+
+interface TargetBasePayload {
+    name: string;
+    devices: string[];
+    when: string;
+    content: {
+        title: string;
+        body: string;
+        platform_specific?: Record<string, any>;
+        replacements?: Record<string, any>;
+    };
+    ttl?: number;
+    wzrk_cid?: string;
+    wzrk_bc?: number;
+    wzrk_bi?: string;
+    system_control_group_include?: boolean;
+    control_group?: CampaignControlGroup | CustomControlGroup;
+    wzrk_acts?: string;
+    respect_throttle?: boolean;
+    send_to_all_devices?: boolean;
+}
+
+type TargetSegmentPayload = {
+    segment: 'all';
+} & TargetBasePayload;
+
+type TargetWherePayload = {
+    where: {
+        event_name?: string;
+        from?: number;
+        to?: number;
+        common_profile_properties?: {
+            profile_fields: Array<Record<string, any>>;
+        };
+    };
+} & TargetBasePayload;
+
+type TargetCreatePayload = TargetWherePayload | TargetSegmentPayload;
+
+interface TargetOptions {
+    debug?: number;
+}
+
+interface clevertap {
+    upload(data: UploadData[], options?: UploadOptions, callback?: Callback): Promise<any>;
+    profile(options: ProfileOptions, callback?: Callback): Promise<any>;
+    profiles(query: Query, options?: QueryOptions, callback?: Callback): Promise<any>;
+    events(query: Query, options?: QueryOptions, callback?: Callback): Promise<any>;
+    targets(
+        action: 'create' | 'estimate',
+        payload: TargetCreatePayload,
+        options?: TargetOptions,
+        callback?: Callback,
+    ): Promise<any>;
+    targets(
+        action: 'list',
+        payload: { from?: number; to?: number },
+        options?: TargetOptions,
+        callback?: Callback,
+    ): Promise<any>;
+    targets(
+        action: 'stop' | 'result',
+        payload: { id: number },
+        options?: TargetOptions,
+        callback?: Callback,
+    ): Promise<any>;
+    TARGET_CREATE: 'create';
+    TARGET_ESTIMATE: 'estimate';
+    TARGET_LIST: 'list';
+    TARGET_RESULT: 'result';
+    TARGET_STOP: 'stop';
+    TARGET_ACTIONS: ['create', 'estimate', 'list', 'result', 'stop'];
+}
+
+declare enum REGIONS {
+    EUROPE = 'eu1',
+    INDIA = 'in1',
+    SINGAPORE = 'sg1',
+    US = 'us1',
+}
+declare function init(accountId: string, accountPasscode: string, region?: REGIONS): clevertap;
```

Comment:
you can document optional param (and other params) here, to say `@default REGIONS.CLEVERTAP_REGIONS.EUROPE`